### ISSUE TITLE
It isn't possible to set 'wildchar' through vim.opt in lua because it expects a number instead. Applies to wildcharm as well.

### ISSUE THREAD

**Issue by jorgesca** [neovim/neovim#18000]
### Neovim version (nvim -v)

0.6.1

### Vim (not Nvim) behaves the same?

No (lua related)

### Operating system/version

x86_64 GNU/Linux

### Terminal name/version

xterm 372

### $TERM environment variable

xterm-256color

### Installation

Arch official community repo

### How to reproduce the issue

echo "vim.opt.wildchar = \"<c-z>\"" > test_init.lua
nvim -u ./test_init.lua

### Expected behavior

Neovim starts and the value of wildchar is set to "<c-z>"

### Actual behavior

Neovim greets with error message upon start:

> Error detected while processing /home/jscandal/test_init.lua:
E5113: Error while calling lua chunk: vim/_meta.lua:324: Invalid option type 'string' for 'wildchar', should be number
stack traceback:
        [C]: in function 'error'
        vim/_meta.lua:324: in function 'assert_valid_value'
        vim/_meta.lua:393: in function 'convert_value_to_vim'
        vim/_meta.lua:681: in function '_set'
        vim/_meta.lua:724: in function '__newindex'
        ./test_init.lua:1: in main chunk
Press ENTER or type command to continue

**Comment by zeertzjq** [comment]
You can use something like `('a'):byte()`.

**Comment by jorgesca** [comment]
> You can use something like `('a'):byte()`.

Thanks, this works! Where's the byte() method coming from? Couldn't find it in the docs. Does it come from lua?

**Comment by zeertzjq** [comment]
> Does it come from lua?

Yes. <https://www.lua.org/manual/5.1/manual.html#pdf-string.byte>

**Comment by smolck** [comment]
What's the thought on how to fix this? Special-case these options or . . . ?

**Comment by justinmk** [comment]
> What's the thought on how to fix this? Special-case these options or . . . ?

That seems reasonable. `vim.o` (formerly `vim.opt`, see https://github.com/neovim/neovim/issues/18246 ) should handle options as users expect.

**Comment by justrajdeep** [comment]
> > You can use something like `('a'):byte()`.
> 
> Thanks, this works! Where's the byte() method coming from? Couldn't find it in the docs. Does it come from lua?

Hi

how do I use the workaround for setting `<TAB>` ?

**Comment by zeertzjq** [comment]
`('\t'):byte()`

**Comment by DominicRacine** [comment]
Where is the \t come from ?

**Comment by zeertzjq** [comment]
It is a Lua escape sequence.
